File: app/src/main/java/at/shockbytes/corey/body/wearable/WearableManager.java
package at.shockbytes.corey.body.wearable;

import android.support.v4.app.FragmentActivity;

import java.util.List;

import at.shockbytes.corey.common.core.workout.model.Workout;

/**
 * @author Martin Macheiner
 *         Date: 18.03.2017.
 */

public interface WearableManager {

    void connect(FragmentActivity activity);

    void synchronizeWorkouts(List<Workout> workouts);

    void onPause();

}

File: app/src/main/java/at/shockbytes/corey/wearable/WearableManager.kt
package at.shockbytes.corey.wearable

import at.shockbytes.corey.common.core.util.WatchInfo
import at.shockbytes.corey.common.core.workout.model.Workout

/**
 * Author:  Martin Macheiner
 * Date:    18.03.2017
 */
interface WearableManager {

    fun onStart(nodeListener: ((WatchInfo) -> Unit)? = null)

    fun onPause()

    fun synchronizeWorkouts(workouts: List<Workout>)
}
